lib/grandpa/message_handler.go

```diff
@@ -79,6 +78,27 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 }
 
 func (h *MessageHandler) handleNeighbourMessage(msg *NeighbourPacketV1) error {
+	if h.grandpa.authority {
+		// TODO: this is a simple hack to ensure that the neighbour messages
```

Comment:
Because we should fix it to don't rely on messages received from other nodes, we should be able to send messages whenever the connection is in a proper state, so I've created the issue and I will link with this TODO